tools/parmetis_wrapper.py

```diff
@@ -0,0 +1,319 @@
+        assert len(tokens) == 3
+
+        src_ntype_name = tokens[0]
+        rel_name = tokens[1]
+        dst_ntype_name = tokens[2]
+
+        data_df = csv.read_csv(edge_info[rank],
+                      read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True),
+                      parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+        data_f0 = data_df['f0'].to_numpy()
+        data_f1 = data_df['f1'].to_numpy()
+
+        global_src_id = data_f0 + ntype_gnid_offset[src_ntype_name][0, 0]
+        global_dst_id = data_f1 + ntype_gnid_offset[dst_ntype_name][0, 0]
+        cols = [global_src_id, global_dst_id]
+        col_names = ["global_src_id", "global_dst_id"]
+        #global_type_eid = np.arange(edge_tids[etype_name][rank][0],\
+        #     edge_tids[etype_name][rank][1] ,dtype=np.int64)
+        #etype_id = etype_name_idmap[etype_name] * \
+        #        np.ones(shape=(data_df['f0'].to_numpy().shape), dtype=np.int64)
+
+        out_file = edge_info[rank].split("/")[-1]
+        out_file = os.path.join(outdir, 'edges_{}'.format(out_file))
+        options = csv.WriteOptions(include_header=False, delimiter=" ")
+        options.delimiter = " "
+
+        csv.write_csv(pyarrow.Table.from_arrays(cols, names=col_names), out_file, options)
+
+        edge_files.append(out_file)
+
+    return edge_files
+
+def read_node_features(schema_map, tgt_ntype_name, feat_names):
+
+    rank, world_size = get_proc_info()
+    node_features = {}
+    if constants.STR_NODE_DATA in schema_map:
+        dataset_features = schema_map[constants.STR_NODE_DATA]
+        if((dataset_features is not None) and (len(dataset_features) > 0)):
+            for ntype_name, ntype_feature_data in dataset_features.items():
+                if ntype_name != tgt_ntype_name : 
+                    continue
+                #ntype_feature_data is a dictionary
+                #where key: feature_name, value: dictionary in which keys are "format", "data"
+                for feat_name, feat_data in ntype_feature_data.items():
+                    if feat_name in feat_names:
+                        my_feat_data_fname = feat_data[constants.STR_DATA][rank]
+                        print('Reading: ', my_feat_data_fname)
+                        if (os.path.isabs(my_feat_data_fname)):
+                            node_features[feat_name] = np.load(my_feat_data_fname)
+                        else:
+                            node_features[feat_name] = np.load(os.path.join(input_dir, my_feat_data_fname))
+    return node_features
+
+
+def gen_node_weights_files(schema_map, output):
+    """
+    Function to create node weight files for ParMETIS along with the edge files.
+
+    Parameters:
+    -----------
+    schema_map : json dictionary
+        dictionary created by reading the metadata.json file for the input dataset
+    output : string
+        location of storing the node-weights and edge files for ParMETIS
+
+    Returns:
```

Comment:
How about adding below for clarity: `<node_type_id> <node_weight_list> <type_wise_node_id>`

Reply:
added the format to the docstring as suggested.